src/main/java/com/uid2/shared/attest/UidCoreClient.java

```diff
@@ -71,179 +59,90 @@ public ICloudStorage getContentStorage() {
-            }
-
-            int statusCode = connection.getResponseCode();
-            notifyResponseStatusWatcher(statusCode);
-
-            if (statusCode < 200 || statusCode >= 300) {
-                LOGGER.warn("attestation failed with UID2 Core returning statusCode=" + statusCode);
-                throw new UidCoreClientException(statusCode, "unexpected status code from uid core service");
-            }
-
-            String response = Utils.readToEnd(connection.getInputStream());
-            JsonObject responseJson = (JsonObject) Json.decodeValue(response);
-            if (isFailed(responseJson)) {
-                throw new UidCoreClientException(statusCode, "response did not return a successful status");
-            }
-
-            String atoken = getAttestationToken(responseJson);
-            if (atoken == null) {
-                throw new UidCoreClientException(statusCode, "response json does not contain body.attestation_token");
-            }
-
-            atoken = new String(decrypt(Base64.getDecoder().decode(atoken), keyPair.getPrivate()), StandardCharsets.UTF_8);
-            LOGGER.info("Attestation successful. Attestation token received.");
-            this.attestationToken.set(atoken);
-        } catch (AttestationException ae) {
-            throw new UidCoreClientException(ae);
-        } catch (IOException ioe) {
-            throw ioe;
-        } catch (Exception e) {
-            throw new UidCoreClientException(e);
+            httpResponse = httpClient.send(httpRequest, HttpResponse.BodyHandlers.ofString());
+        } catch (IOException | InterruptedException e) {
+            LOGGER.error("Failed to send request with error: ", e);
+            throw e;
         }
+        return httpResponse;
     }
 
-    private static boolean isFailed(JsonObject responseJson) {
-        return responseJson.getString("status") == null || !responseJson.getString("status").equals("success");
-    }
-
-    private static String getAttestationToken(JsonObject responseJson) {
-        final JsonObject body = responseJson.getJsonObject("body");
-        if(body == null) return null;
-        return body.getString("attestation_token");
-    }
-
-    private static byte[] decrypt(byte[] payload, PrivateKey privateKey) throws Exception {
-        Cipher cipher = Cipher.getInstance(Const.Name.AsymetricEncryptionCipherClass);
-        cipher.init(Cipher.DECRYPT_MODE, privateKey);
-        return cipher.doFinal(payload);
+    public void setResponseStatusWatcher(Handler<Integer> watcher) {
+        this.responseWatcher = watcher;
     }
 
-    private static KeyPair generateKeyPair() throws NoSuchAlgorithmException {
-        KeyPairGenerator gen = KeyPairGenerator.getInstance(Const.Name.AsymetricEncryptionKeyClass);
-        gen.initialize(2048, new SecureRandom());
-        return gen.generateKeyPair();
+    public void setAttestationTokenRetriever(AttestationTokenRetriever attestationTokenRetriever) {
```

Comment:
It's better to have the passed in on the constructor, rather than being able to change it.